cdcpd/tests/integration/static_rope.cpp

```diff
@@ -0,0 +1,142 @@
+#include <cdcpd/cdcpd.h>
+
+#include <ros/ros.h>
+#include <rosbag/bag.h>
+#include <rosbag/view.h>
+#include <pcl/point_types.h>
+#include <pcl_ros/point_cloud.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <string>
+#include <gtest/gtest.h>
+
+#include <iostream>
+
+#include "cdcpd/utils.h"
+#include "test_resim_utils.h"
+
+#define PRINT_DEBUG_MESSAGES false
+
+// NOTE: This is refactored code so that I can get the initial tracked points more than once to
+// initialize the tracked points that are updated over time.
+std::pair<PointCloud::Ptr, Eigen::Matrix2Xi const> getInitialTracking(
+    float const max_rope_length, int const num_points)
+{
+    Eigen::Vector3f start_position{Eigen::Vector3f::Zero()};
+    Eigen::Vector3f end_position{Eigen::Vector3f::Zero()};
+    start_position << -max_rope_length / 2, 0, 1.0;
+    end_position << max_rope_length / 2, 0, 1.0;
+    std::pair<Eigen::Matrix3Xf, Eigen::Matrix2Xi> const& initial_template_pair =
+        makeRopeTemplate(num_points, start_position, end_position);
+    Eigen::Matrix3Xf const& initial_template_vertices = initial_template_pair.first;
+    Eigen::Matrix2Xi const initial_template_edges = initial_template_pair.second;
+    pcl::PointCloud<pcl::PointXYZ>::Ptr initial_tracked_points =
+        makeCloud(initial_template_vertices);
+    return {initial_tracked_points, initial_template_edges};
+}
+
+CDCPD* initializeCdcpdSimulator(PointCloud::Ptr const& initial_tracked_points,
+    Eigen::Matrix2Xi initial_template_edges, float const max_rope_length, int const num_points)
+{
+    if (PRINT_DEBUG_MESSAGES)
+    {
+        std::stringstream ss;
+        ss << initial_template_edges;
+        ROS_WARN("Initial template edges");
+        ROS_WARN(ss.str().c_str());
+    }
+
+    // Specify constant scalar and boolean parameters
+    float const objective_value_threshold = 1.0;
+    bool const use_recovery = false;
+    double const alpha = 0.5;
+    double const beta = 1.0;
+    double const lambda = 1.0;
+    double const k = 100.0;
+    float const zeta = 10.0;
+    float const obstacle_cost_weight = 0.001;
+    float const fixed_points_weight = 10.0;
+
+    CDCPD* cdcpd_ptr = new CDCPD(initial_tracked_points, initial_template_edges,
```

Comment:
`new` without `delete` is a memory leak. I know it doesn't matter because this is just a quick test, but I think we can do better. Instead, just return the `CDCPD` object by value, and let the compiler do `RVO`. Alternatively, use a `std::shared_ptr`